.travis.yml

```diff
@@ -11,21 +11,26 @@ env:
 before_install:
   - chmod +x gradlew
 
-after_success:
-  # See: https://github.com/codecov/example-java/blob/master/.travis.yml
-  - bash <(curl -s https://codecov.io/bash)
-  - chmod +x ./scripts/report-coverage.sh
-  - ./scripts/report-coverage.sh
-  - chmod +x ./scripts/publish-artifacts.sh
-  - ./scripts/publish-artifacts.sh
-
 script:
   - ./gradlew check --stacktrace
+
   # Decrypt and unarchive Maven and Google Cloud Storage credentials.
   - openssl aes-256-cbc -K $encrypted_98c791cdf535_key -iv $encrypted_98c791cdf535_iv -in credentials.tar.enc -out credentials.tar -d
   - tar xvf credentials.tar
+
   - ./gradlew cleanGCS --info
 
+  # The publishing script should be executed in `script` section in order to
+  # fail the Travis build if execution of this script is failed.
+  - chmod +x ./scripts/publish-artifacts.sh
+  - ./scripts/publish-artifacts.sh
```

Comment:
@dmytro-grankin Please check if it's going to be executed even if previous phases fail (e.g. compilation failure occurred). It should not.